### PR TITLE
PDF Previewer user profile option

### PR DESCRIPTION
**Pull request by pferreir**

### CODE REVIEW

File: indico/modules/attachments/preview.py [hunk 1]
```diff
@@ -57,6 +57,11 @@ class PDFPreviewer(Previewer):
     ALLOWED_CONTENT_TYPE = re.compile(r'^application/pdf$')
     TEMPLATE = 'iframe_preview.html'
 
+    @classmethod
+    def can_preview(cls, attachment_file):
+        return (super(PDFPreviewer, cls).can_preview(attachment_file) and
+                session.user.settings.get('use_previewer_pdf', False))
```
Comment (ThiefMaster) [inline indico/modules/attachments/preview.py]: `session.user` may be `None`

File: indico/modules/users/forms.py [hunk 2]
```diff
@@ -48,10 +48,14 @@ class UserPreferencesForm(IndicoForm):
     timezone = SelectField(_('Timezone'))
     force_timezone = BooleanField(_('Use my timezone'),
                                   widget=SwitchWidget(),
-                                  description='Always use my current timezone instead of an event\'s timezone.')
+                                  description=_('Always use my current timezone instead of an event\'s timezone.'))
```
Comment (ThiefMaster) [inline indico/modules/users/forms.py]: double quotes around the string instead of escaping '`'?

File: indico/modules/users/forms.py [hunk 3]
```diff
@@ -48,10 +48,14 @@ class UserPreferencesForm(IndicoForm):
...
     show_past_events = BooleanField(_('Show past events'),
                                     widget=SwitchWidget(),
-                                    description='Show past events by default.')
+                                    description=_('Show past events by default.'))
+    use_previewer_pdf = BooleanField(
+        _('Use previewer for PDF files'),
```
Comment (ThiefMaster) [inline indico/modules/users/forms.py]: different style than what we use for all the other form fields

File: indico/modules/users/forms.py (hunk 2 above)
Comment (pferreir) [inline indico/modules/users/forms.py]: It was already like that. Yeah, I'll take the opportunity to fix it.

File: indico/modules/users/forms.py [hunk 4]
```diff
@@ -48,10 +48,14 @@ class UserPreferencesForm(IndicoForm):
...
+        widget=SwitchWidget(),
+        description=_('The previewer is used by default for image and text files, but not for PDFs.'))
```
Comment (ThiefMaster) [inline indico/modules/users/forms.py]: "pdf files" might sound better

File: indico/modules/attachments/preview.py (hunk 1 above)
Comment (pferreir) [inline indico/modules/attachments/preview.py]: Oh, :hankey:. What should we use by default? I guess `False`.

Comment (ThiefMaster) [inline indico/modules/attachments/preview.py]: yeah, i guess it makes sense to use the same default for both guests and users

**Comment by ThiefMaster** [comment]
Didn't we also want to completely disable the previewer for on mobile devices? Might be a good opportunity to do this now, too.

File: indico/modules/users/forms.py (hunk 3 above)
Comment (pferreir) [inline indico/modules/users/forms.py]: There's a clear lack of space for the whole line. Are you suggesting I change it in all of them?

**Comment by pferreir** [comment]
@ThiefMaster any suggestions on how to do that. Shall we check the UserAgent string directly?

File: indico/modules/users/forms.py (hunk 3 above)
Comment (ThiefMaster) [inline indico/modules/users/forms.py]:
```
     use_previewer_pdf = BooleanField(_('Use previewer for PDF files'), widget=SwitchWidget(),
                                      description=_('The previewer is used by default for image and text files, but '
                                                    'not for PDF files.'))
```

would have been my choice.. maybe more a matter of taste

**Comment by ThiefMaster** [comment]
client-side, `$.mobileBrowser` (that's what we use for the indicomobile message)

**Comment by ThiefMaster** [comment]
Might also be a good idea to use the latest regex from http://detectmobilebrowsers.com/ in case it changed since we added it

**Comment by pferreir** [comment]
Done.

File: indico/modules/attachments/preview.py [hunk 5]
```diff
@@ -57,6 +57,11 @@ class PDFPreviewer(Previewer):
     ALLOWED_CONTENT_TYPE = re.compile(r'^application/pdf$')
     TEMPLATE = 'iframe_preview.html'
 
+    @classmethod
+    def can_preview(cls, attachment_file):
+        use_previewer = session.user.settings.get('use_previewer_pdf', False) if session.user else False
+        return (super(PDFPreviewer, cls).can_preview(attachment_file) and use_previewer)
```
Comment (mic4ael) [inline indico/modules/attachments/preview.py]: Are the brackets needed?

Comment (ThiefMaster) [inline indico/modules/attachments/preview.py]: No, but it used to be on two lines and I guess somebody forgot to remove them

**Pull request merged**